WordPress/gutenberg
# JSON Schema Docgen Rework

Reopened #63639 so that it could be merged in the correct order after #63591.

> ## What?
> 
> Reworks the theme.json reference docgen.
> 
> Key features include:
> 
> * Descriptions for styles and settings properties.
> * Reworked type serialization to remove the props heading for styles and settings.
> * Better types everywhere in the document.
> * Details for settings and styles.
> * The useRootPaddingAwareAlignments top-level property returns.
> * Cleaner code.
> 
> ## Why?
> 
> Split out of #63591 so that both the schema changes and codegen changes can be reviewed independently from one another.
> ## How?
> 
> Slow refactoring of the docgen code bit by bit. Check out the commits for details.
> ## Testing Instructions
> 
> See that the [new docs](https://github.com/WordPress/gutenberg/blob/update/json-schema-codegen-rework/docs/reference-guides/theme-json-reference/theme-json-living.md?rgh-link-date=2024-07-17T02%3A47%3A59Z) look correct.
> ### Testing Instructions for Keyboard
> 
> N/A
> ## Screenshots or screencast
> 
> N/A
